tests/e2e/multi_vc_utils.go

```diff
@@ -0,0 +1,433 @@
+			accessMode
+		statefulset.Name = stsName
+		statefulset.Spec.Template.Labels["app"] = statefulset.Name
+		statefulset.Spec.Selector.MatchLabels["app"] = statefulset.Name
+	}
+	if nodeAffinityToSet {
+		nodeSelectorTerms := getNodeSelectorTerms(allowedTopologies)
+		statefulset.Spec.Template.Spec.Affinity = new(v1.Affinity)
+		statefulset.Spec.Template.Spec.Affinity.NodeAffinity = new(v1.NodeAffinity)
+		statefulset.Spec.Template.Spec.Affinity.NodeAffinity.
+			RequiredDuringSchedulingIgnoredDuringExecution = new(v1.NodeSelector)
+		statefulset.Spec.Template.Spec.Affinity.NodeAffinity.
+			RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms = nodeSelectorTerms
+	}
+	if podAntiAffinityToSet {
+		statefulset.Spec.Template.Spec.Affinity = &v1.Affinity{
+			PodAntiAffinity: &v1.PodAntiAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+					{
+						LabelSelector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								"key": "app",
+							},
+						},
+						TopologyKey: "topology.kubernetes.io/zone",
+					},
+				},
+			},
+		}
+
+	}
+	if isParallelPodMgmtPolicy {
+		statefulset.Spec.PodManagementPolicy = appsv1.ParallelPodManagement
+	}
+	statefulset.Spec.Replicas = &replicas
+
+	framework.Logf("Creating statefulset")
+	CreateStatefulSet(namespace, statefulset, client)
+
+	framework.Logf("Wait for StatefulSet pods to be in up and running state")
+	fss.WaitForStatusReadyReplicas(client, statefulset, replicas)
+	gomega.Expect(fss.CheckMount(client, statefulset, mountPath)).NotTo(gomega.HaveOccurred())
+	ssPodsBeforeScaleDown := fss.GetPodList(client, statefulset)
+	gomega.Expect(ssPodsBeforeScaleDown.Items).NotTo(gomega.BeEmpty(),
+		fmt.Sprintf("Unable to get list of Pods from the Statefulset: %v", statefulset.Name))
+	gomega.Expect(len(ssPodsBeforeScaleDown.Items) == int(replicas)).To(gomega.BeTrue(),
+		"Number of Pods in the statefulset should match with number of replicas")
+
+	return statefulset
+}
+
+/*
+setSpecificAllowedTopology returns topology map with specific topology fields and values
+in a multi-vc setup
+*/
+func setSpecificAllowedTopology(allowedTopologies []v1.TopologySelectorLabelRequirement,
+	topkeyStartIndex int, startIndex int, endIndex int) []v1.TopologySelectorLabelRequirement {
+	var allowedTopologiesMap []v1.TopologySelectorLabelRequirement
+	specifiedAllowedTopology := v1.TopologySelectorLabelRequirement{
+		Key:    allowedTopologies[topkeyStartIndex].Key,
+		Values: allowedTopologies[topkeyStartIndex].Values[startIndex:endIndex],
+	}
+	allowedTopologiesMap = append(allowedTopologiesMap, specifiedAllowedTopology)
+
+	return allowedTopologiesMap
+}
+
+/*
+If we have multiple statefulsets, deployment Pods, PVCs/PVs created on a given namespace and for performing
+cleanup of these multiple sts creation, deleteAllStatefulSetAndPVs is used
+*/
+func deleteAllStatefulSetAndPVs(ctx context.Context, c clientset.Interface, ns string) {
```

Comment:
Please give appropriate method name (suggestion to use: deleteAllStsAndDeployInNamespace)

Reply:
Ack, took this name - deleteAllStsAndPodsPVCsInNamespace